src/sentry/db/models/query.py

```diff
@@ -49,6 +66,50 @@ def update(self: Model, using: str | None = None, **kwargs: Any) -> int:
 update.alters_data = True  # type: ignore
 
 
+def update_or_create(
```

Comment:
One difference between this and the old function is that we used to be able to differentiate between defaults and values for updating. So defaults would only be used on create, where values would be used on both create/update.

I'm not sure if we have any code that relies on that behaviour. If not, then I'm find with dropping it, but let's double check.

Reply:
OK I double checked and the only calls that were using both `values` and `defaults` were:
```py
GroupEnvironment.objects.create_or_update(
    environment_id=caches["Environment"](project.organization_id, env_name).id,
    group_id=group_id,
    defaults=fields,
    values=fields,
)
```

```py
rows_affected, created = OrganizationOnboardingTask.objects.create_or_update(
    organization=organization, task=task_id, values=values, defaults={"user": request.user}
)
```
```py
OrganizationMemberTeam.objects.create_or_update(
    organizationmember=to_member, team=omt.team, defaults={"is_active": True}
)
```
The first one is equivalent to omitting `defaults`, the second one just for the product tutorial, and the third one should probably be replaced with a `create()`.

Reply:
For the second this does change behaviour, but maybe it's not too important...